packages/svelte/src/internal/client/dom/elements/misc.js

```diff
@@ -56,3 +57,14 @@ export function add_form_reset_listener() {
 		);
 	}
 }
+
+/**
+ * @param {string} text
+ */
+export function title(text) {
+	const previous = document.title;
+	document.title = text;
+	teardown(() => {
+		document.title = previous;
+	});
```

Comment:
I haven't been able to break this, but it _feels_ like we should put the setup in an effect so that there's no possibility of another component setting the title to a new value immediately before the teardown nukes it. might just be superstition

```suggestion
	const previous = document.title;

	effect(() => {
		document.title = text;

		return () => {
			document.title = previous;
		};
	});
```

Reply:
The effect does nothing, since text is a static value; I kept the "add document.title = .. to the shared effect" part within the compiler which is why it's updating still. The assumption is that it's rare that people have a reactive title. But we can move it into here

Reply:
It's not about reactivity, it's about preventing this sequence of events from being a possibility:

1. component A is removed and component B is created at the same moment
2. component B sets `document.title`
3. component A is removed and reverts `document.title` to whatever it was before `A` was created

That may already be impossible, hence 'superstition'

Reply:
In which way does adding a `render_effect` help with this? The render effect is called synchronously, and `teardown` is a render effect under the hood.

Reply:
was more of a belt-and-braces thing — see https://github.com/sveltejs/svelte/pull/14116#issuecomment-2455754219. but it also means that if/when we introduce forking, we won't set the title until the fork is committed

Reply:
Yeah, this isn't a `render_effect`, it's an `effect`, so it won't run in the fork until it's committed like Rich says.

Reply:
Well the code that was committed is clearly a render effect

Reply:
Sorry, I was going on about the above suggestion. If it's a render_effect, then that's not right.

Reply:
Surely render effects other than blocks wouldn't run until forks are committed? The whole point is to not update the DOM until then

Reply:
Maybe. Too early to say for sure how that will all end up. Let's just keep it as is then.

Reply:
But keeping it as is currently, i.e. using a `render_effect` for the thing instead of just a `teardown` does nothing, as I've just explained.

Not gonna die on this hill, maybe I don't understand what this proofs against exactly, and using teardown is a microoptimization regardless.

Reply:
yeah, no-one is updating the title at 60fps. given that there's _potential_ upside and no real downside i reckon we can just keep things as they are here — this is ready to merge from my perspective